### PR TITLE
Turn on 1.38 CI

### PR DESCRIPTION
**Pull request by RhinosF1**
Not long at all

### CODE REVIEW

**Comment by Universal-Omega** [comment]
REL1_38 will be added when cut. Also I will likely do this when I work on 1.38 compatability, as it is broken right now I think.

**Comment by RhinosF1** [comment]
> REL1_38 will be added when cut. Also I will likely do this when I work on 1.38 compatability, as it is broken right now I think.

I created the PR to check it works on master and see how far off the CI is. I intend to add 1.38 checks to all our repos based on the branch on Tuesday when preparing our branch.

My aim as always is to get 1.38 live as soon as possible (from Thursday evening is safe waters).

**Comment by Universal-Omega** [comment]
> I created the PR to check it works on master and see how far off the CI is.

Even marked as experimental you can check the workflow manually, it will tell you what's off, just won't fail the overall CI.

**Comment by Universal-Omega** [comment]
> That won't stop people breaking it more once its passing

Making it fail the CI makes no difference about how its broken in CentralAuth. Unmarking experimental makes absolutely no difference to how the CI runs, only causes the overall workflow to fail. All it does is sets `continue-on-error`.

File: includes/SpecialRemovePII.php [hunk 1]
```diff
@@ -15,7 +15,6 @@
 use MediaWiki\Extension\CentralAuth\GlobalRename\GlobalRenameUserValidator;
 use MediaWiki\Extension\CentralAuth\User\CentralAuthUser;
 use MediaWiki\JobQueue\JobQueueGroupFactory;
-use MediaWiki\User\UserFactory;
```
Comment (Universal-Omega) [inline includes/SpecialRemovePII.php:17]: Why remove this? You should remove CentralAuthUser (the non-namespaced one)

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Comment by RhinosF1** [comment]
> > That won't stop people breaking it more once its passing
> 
> Making it fail the CI makes no difference about how its broken in CentralAuth. Unmarking experimental makes absolutely no difference to how the CI runs, only causes the overall workflow to fail. All it does is sets `continue-on-error`.

But when I make it green, it will all pass

**Comment by Universal-Omega** [comment]
> > > That won't stop people breaking it more once its passing
> > 
> > 
> > Making it fail the CI makes no difference about how its broken in CentralAuth. Unmarking experimental makes absolutely no difference to how the CI runs, only causes the overall workflow to fail. All it does is sets `continue-on-error`.
> 
> But when I make it green, it will all pass

Yes, but we still (even if passing) want master to be experimental. It is breaking to much, and not need to be bothered by until release approaches.

Also before this is even made to simply pass, there is alot of work I want to do to better modernise the code as well. New services from CentralAuth need to be injected to the special page is one thing that needs done... In addition, doing this will only make REL1_37 CI fail instead.

File: includes/SpecialRemovePII.php (hunk 1 above)
Comment (RhinosF1) [reply includes/SpecialRemovePII.php:17]: I thought I added the namespaced one

Comment (Universal-Omega) [reply includes/SpecialRemovePII.php:17]: You did. You need to remove the first one or else it is a confliction.

File: includes/SpecialRemovePII.php [hunk 2]
```diff
@@ -133,7 +134,7 @@ public function validate( array $formData ) {
 			return Status::newFatal( 'centralauth-rename-doesnotexist' );
 		}
 
-		$oldCentral = CentralAuthUser::getInstanceByName( $formData['oldname'] );
+		$oldCentral = \CentralAuthUser::getInstanceByName( $formData['oldname'] );
```
Comment (Universal-Omega) [inline includes/SpecialRemovePII.php:137]: Way that? Thus just makes the current behaviour, as if no `use` statement was added.

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Comment by RhinosF1** [comment]
If we can't make both pass, there need to be release branches.

I can not merge the experimental change but it still make it slightly easier for me to see is it failing

File: includes/SpecialRemovePII.php (hunk 2 above)
Comment (RhinosF1) [reply includes/SpecialRemovePII.php:137]: Based it on the only other change in the equivalent PR for our other extensions to see if it did anything to help it pass.

**Pull request closed without merging**